tests/tests_system/test_process.py

```diff
@@ -26,7 +26,12 @@ def test(self) -> None:
 
         return_code = process_handler.terminate()
 
-        self.assertIsNone(return_code)
+        # In Python 3.9 it seems that Process Termination is not as stable in 3.7.
```

Comment:
Will this lead to process being left in running state if termination is not successful? Maybe create an issue for this too to address it later?

Reply:
This is not related to the actual termination where a wait will happen until proper termination.
The test is trying to close a read call, that is always failing on 3.7 consistently, thus always `None`.

In 3.9 on some OS like ubuntu it is still observe as `None`, while on macos and CentOS7 I have observed `1` which is correctly termination of the process.

This is so inconsistent that even adding a 10seconds timer would not resolve this.

We probably will revisit this but this is more like a flaky test than a problematic implementation on the actual code. It is just it always fail on 3.7 but sometimes run correctly on 3.9. Thanks.

Reply:
Thanks maybe open a bug? We can take help from community too.